tests/test_coding_apply_blocks.py

```
@@ -0,0 +1,173 @@
+    (ws_dir / "main.py").write_text("old content\n")
+    r = await client.post(
+        f"/api/coding/workspaces/{ws_id}/apply-blocks",
+        json={"blocks": [{"path": "main.py", "content": "new content\n"}]},
+    )
+    assert r.status_code == 200, r.text
+    assert (ws_dir / "main.py").read_text() == "new content\n"
+
+
+@pytest.mark.asyncio
+async def test_apply_blocks_shows_in_diff(ws):
+    client, ws_id, ws_dir = ws
```

Ruff (0.15.17)

[warning] Unpacked variable `ws_dir` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)